src/flist/flist_debian_case_study.md

```diff
@@ -0,0 +1,290 @@
+
+For this case study, we draw inspiration from the [Ubuntu 22.04](https://github.com/threefoldtech/tf-images/tree/development/tfgrid3/ubuntu22.04) directory.
+
+If we look at the Ubuntu 22.04 directory tree, this is what we get:
+
+```
+.
+├── Dockerfile
+├── README.md
+├── start.sh
+└── zinit
+    ├── ssh-init.yaml
+    └── sshd.yaml
+```
+
+We will now explore each of those files to get a good look at the whole repository and try to understand how it all works together.
+
+### Dockerfile
+
+We recall that to make a Docker image, you need to create a Dockerfile. As per [Docker's documentation](https://docs.docker.com/engine/reference/builder/), a Dockerfile is "a Dockerfile is a text document that contains all the commands a user could call on the command line to assemble an image". 
+
+The Ubuntu 22.04 Dockerfile is as follows:
+
+File: `Dockerfile`
+
+```Dockerfile
+FROM ubuntu:22.04
+
+RUN apt update && \
+  apt -y install wget openssh-server
+
+RUN wget -O /sbin/zinit https://github.com/threefoldtech/zinit/releases/download/v0.2.5/zinit && \
+  chmod +x /sbin/zinit
+
+COPY zinit /etc/zinit
+COPY start.sh /start.sh
+
+RUN chmod +x /sbin/zinit && chmod +x /start.sh
+ENTRYPOINT  ["zinit", "init"]
+```
+
+We can see from the first line that the Dockerfile will look for the docker image `ubuntu:22.04`. In our case, we want to get the Debian 12 docker image. This information is available on the Docker Hub (see [Debian Docker Hub](https://hub.docker.com/_/debian)).
+
+We will thus need to change the line `FROM ubuntu:22.04` to the line `FROM debian:12`. It isn't more complicated than that!
+
+We now have the following Dockerfile fore the Debian docker image:
+
+File: `Dockerfile`
+
+```Dockerfile
+FROM debian:12
+
+RUN apt update && \
+  apt -y install wget openssh-server
+
+RUN wget -O /sbin/zinit https://github.com/threefoldtech/zinit/releases/download/v0.2.5/zinit && \
+  chmod +x /sbin/zinit
+
+COPY zinit /etc/zinit
+COPY start.sh /start.sh
+
+RUN chmod +x /sbin/zinit && chmod +x /start.sh
+ENTRYPOINT  ["zinit", "init"]
+```
+
+There is nothing more needed here. Pretty fun to start from existing open-source code, right?
+
+### Docker Image Script
+
+The other important file we will be looking at is the `start.sh` file. This is the basic script that will be used to properly set the docker image. Thankfully, there is nothing more to change in this file, we can leave it as is. As we will see later, this file will be executed by zinit when the container starts.
+
+File: `start.sh`
+
+```.sh
+#!/bin/bash
+
+mkdir -p /var/run/sshd
+mkdir -p /root/.ssh
+touch /root/.ssh/authorized_keys
+
+chmod 700 /root/.ssh
+chmod 600 /root/.ssh/authorized_keys
+
+echo "$SSH_KEY" >> /root/.ssh/authorized_keys
+```
+
+### zinit Folder
+
+Next, we want to take a look at the zinit folder. When we start the Docker container, the files in the folder zinit will be executed.
+
+If we take a look at the file `ssh-init.yaml`, we find the following:
+
+```.yaml
```

Comment:
in this zinit service file, we define a service named `ssh-init.yaml`, we tell zinit which command to execute `bash /start.sh`, where to log, here in the stdout and it's a `oneshot`, meaning to be executed only once